routing/pathfind.go

```diff
@@ -142,6 +156,17 @@ func newRoute(sourceVertex route.Vertex,
 			totalTimeLock += uint32(finalHop.cltvDelta)
 
 			outgoingTimeLock = currentHeight + uint32(finalHop.cltvDelta)
+
+			// If we have a feature vector for the final hop, we'll
+			// determine if they support TLV onion payloads by
+			// checking if a var_onion_optin feature is set. If we
+			// already know the hop supports TLV from their node
+			// announcement, we can skip this step.
+			if !tlvPayload && finalHop.features != nil {
+				tlvPayload = finalHop.features.HasFeature(
+					lnwire.TLVOnionPayloadOptional,
+				)
+			}
```

Comment:
What if the graph says yes, but the invoice says no?

Reply:
good point, i think you're right, the invoice should always override the graph.

Reply:
> The graph is telling me noooo, but the invoice, the invoice...is telling me...to deliver a single hop mpp payment with the adequate set of records contained in the TLV payload